system/doc/design_principles/distributed_applications.md

```diff
@@ -169,14 +181,67 @@ than `cp3`, `myapp` is restarted on `cp2`.
 
 [](){: #dist2 }
 
-![Application myapp - Situation 2](assets/dist2.gif "Application myapp - Situation 2")
+```mermaid
+---
+title: Application myapp - Situation 2
+---
+
+stateDiagram-v2
+        Crash --> Failover: 5 secs.
+
+    state Crash {
+        state "myapp" as myappCrash {
```

Comment:
Again I think it makes more sense to have `myapp` inside cp1.

Also, I think the `Crash` and `Failover` states should be called `Before` and `After`.

Reply:
Is `Restart` and `Takeover` for situation 5 fine?

Reply:
I think it should be before also there